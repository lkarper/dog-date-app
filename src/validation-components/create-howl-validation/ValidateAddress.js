import React, { useEffect } from 'react';

const ValidateAddress = (props) => {

    const { address, addressError, setAddressError } = props;

    useEffect(() => {

        if (address.trim().length === 0) {
            setAddressError(`You must enter an address or location name.`);
        } else {
            setAddressError('');
        }

    }, [address, setAddressError]);

    if (addressError) {
        return <p id='address-validator'>{addressError}</p>;
    }

    return <p id='address-validator'>Address/location set to {address}.</p>;
}

export default ValidateAddress;