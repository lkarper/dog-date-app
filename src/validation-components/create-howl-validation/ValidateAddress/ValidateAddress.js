import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

const ValidateAddress = (props) => {

    const { 
        address, 
        addressError, 
        setAddressError 
    } = props;

    useEffect(() => {

        if (address.trim().length === 0) {
            setAddressError(`You must enter an address or location name. (Max 100 chars.)`);
        } else {
            setAddressError('');
        }

    }, [address, setAddressError]);

    if (addressError) {
        return (
            <p 
                className='ValidateAddress__validator error'
                id='address-validator'
            >
                {addressError}
            </p>
        );
    }

    return (
        <p 
            className='ValidateAddress__validator valid'
            id='address-validator'
        >
            Address/location set to '{address}'. (Max 100 chars.)
        </p>
    );
}

ValidateAddress.defaultProps = {
    address: '',
    addressError: `You must enter an address or location name. (Max 100 chars.)`,
    setAddressError: () => {},
};

ValidateAddress.propTypes = {
    address: PropTypes.string.isRequired,
    addressError: PropTypes.string.isRequired,
    setAddressError: PropTypes.func.isRequired,
};

export default ValidateAddress;
