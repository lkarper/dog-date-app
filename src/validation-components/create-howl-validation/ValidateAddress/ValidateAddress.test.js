import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ValidateAddress from './ValidateAddress';

describe('ValidateAddress component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ValidateAddress />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<ValidateAddress />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
