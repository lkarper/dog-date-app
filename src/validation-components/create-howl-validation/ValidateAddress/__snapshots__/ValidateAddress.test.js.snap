// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidateAddress component renders the UI as expected 1`] = `
<ValidateAddress
  address=""
  addressError="You must enter an address or location name. (Max 100 chars.)"
  setAddressError={[Function]}
>
  <p
    className="ValidateAddress__validator error"
    id="address-validator"
  >
    You must enter an address or location name. (Max 100 chars.)
  </p>
</ValidateAddress>
`;
